iati/core/metadata.py

```diff
@@ -0,0 +1,82 @@
+"""A module containing metadata classes for IATI SSOT objects."""
+
+
+class Metadata(object):
+    """A base metadata class for all types of IATI SSOT objects."""
+
+    def __init__(self, **kwargs):
+        """Set class attributes."""
+        self.version = kwargs.pop('version', None)
+        self.orginality = kwargs.pop('orginality', None)
+        self.title = kwargs.pop('title', None)
+        self.description = kwargs.pop('description', None)
+
+
+class MetadataCodelist(Metadata):
+    """A metadata class for codelist.Codelist objects."""
+
+    def __init__(self, **kwargs):
+        """Set class attributes."""
+        self.category_codelist = kwargs.pop('category_codelist', None)
+        self.codelist_name = kwargs.pop('codelist_name', None)
+        self.complete = kwargs.pop('complete', None)
+        self.ref = kwargs.pop('ref', None)
+        self.revision = kwargs.pop('revision', None)
+        self.source = kwargs.pop('source', None)
+        self.type = kwargs.pop('type', None)
+        self.url = kwargs.pop('url', None)
+        Metadata.__init__(self, **kwargs)
```

Comment:
The call to `Metadata.__init__` should be before setting the other values so that nothing in the child is accidentally overwritten by the parent.

(same for other classes)